Add image stamping customization functionality

nirvn opened the pull request:
This PR adds image stamping customization functionality, available in the project properties' QField panel. First, screenshot time:

<img width="734" height="689" alt="image" src="https://github.com/user-attachments/assets/577ed4be-ad68-4baf-8bf6-161f422d53fd" />

As the screenshot shows above, the stamping customization widgets are offered in a sub-panel which can be opened by click on its settings button within the attachments tab here:

<img width="734" height="689" alt="image" src="https://github.com/user-attachments/assets/e736496e-2a44-4958-a583-155fc34d07b3" />

I'm quite happy about the introduction of a panels stack here, it should give us plenty of space to grow in the future.

The image stamping customization allows for users to:
- customize the details being stamped via the crafting of an expression-based multiline string
- customize the font color/size/drop shadow/etc. used to render the details
- customize the position (left / center / right) of the details
- add an image decoration overlay on top of the image being stamped
- force stamping of images irrespective of QField's own setting (useful for manager distributing over QFC)

As a result, people can come up with this kind of customized stamping with just a little bit of configuration:

![apiary_20250715152419984](https://github.com/user-attachments/assets/6d01726f-d929-4da9-a46a-29ac733829ac)

_Note the top-left logo and the QFC username detail_

suricactus commented:
Nice functionality and UI! Two comments here:

1) I don't see if we can configure where the stamping happens: top, bottom, left or right
2) I guess you used a textbox for the expression builder just to be able to see the whole expression at once, but do we really want to diverge from the already existing expression editor UI in QGIS? I don't have a strong opinion against the apporach, but IMO we benefit when we apply the same UI indications everywhere:
<img width="459" height="34" alt="image" src="https://github.com/user-attachments/assets/b9d3f4d4-bbe9-47e2-a2c8-b3bef854e762" />

suricactus reviewed (approved):
Code looks good, two small comments.

nirvn commented:
@suricactus ,

> I don't see if we can configure where the stamping happens: top, bottom, left or right

There's no top / bottom option, only horizontal's left/center/right alignment (via the combobox)

> I guess you used a textbox for the expression builder just to be able to see the whole expression at once, but do we really want to diverge from the already existing expression editor UI in QGIS? I don't have a strong opinion against the approach, but IMO we benefit when we apply the same UI indications everywhere...

What's there is *not* an expression in itself, it's a text where we can insert one (or more) expression within [% %] delimiters. This makes it much easier for people to build long, multiline strings instead of forcing a single expression. This also requires _zero_ expression knowledge if you just want to add a non-data driven stamp (e.g. "Photo taken by Kebab Co., Ltd.") as you just type in your string.

Regarding 'everywhere else', it aligns with what we have in QGIS for layout item label widgets and title / copyright decoration dialogs:

<img width="870" height="571" alt="image" src="https://github.com/user-attachments/assets/c6c4363d-d3aa-4aa1-903f-99e73e4bc739" />

<img width="711" height="292" alt="image" src="https://github.com/user-attachments/assets/50a5bde0-12ce-45a0-9a38-b3d504c5b1d1" />

> I tried it locally and I got a bit confused why my expression was empty, but I still got a preview. It is the default one, I know, but we better show it prepopulated, we already do that with attachment naming expression?

The difference between the attachment naming expression and this is that I'd rather not save a default one and have project glued to the current default when we update it in QField itself. But I can work around that, consider your request adopted sir ;)

> It is not clear the preview is about the expression evaluation, they are a bit distant in space. Probably a better label above it? "Preview expression as text"? 

OK, I'll give something else a try there.

nirvn commented on qfieldsync/gui/image_stamping_configuration_widget.py:
```diff
@@ -0,0 +1,289 @@
+        self.customFontStyleButton.setShowNullFormat(True)
+        self.customFontStyleButton.setNoFormatString(self.tr("Default font style"))
+
+        self.customAlignmentComboBox.addItem(self.tr("Left"))
+        self.customAlignmentComboBox.addItem(self.tr("Center"))
+        self.customAlignmentComboBox.addItem(self.tr("Right"))
```
We can define them in the Qt .ui file or in the .py programmatically. However, I think it's much better practice to define them in the .py file.

nirvn commented:
@suricactus , alright, comments addressed.

First, the slightly tweaked GUI:

<img width="809" height="714" alt="image" src="https://github.com/user-attachments/assets/7e7f8b20-c5e0-43eb-9fe0-e10f9063dfd9" />

As you can see, I've relocated (and renamed) the details ~~expression~~ template to sit in its own group box, where the preview label lives. It makes for a much clearer widget distribution. A default template is also added when opening the subpanel as per your suggestion.

The 'text horizontal alignment' label was renamed to 'horizontal alignment' as it controls both the text and the image.
